### PR TITLE
feat(eighth): add ability to set 'signup_time' on eighth block creation 

### LINKED ISSUE #1097: Add ability to set `EighthBlock` `signup_time` upon creation
## Feature description
On the block creation page, there is no option to set the signup time for blocks. One has to manually go into the edit page for each block and set it.

## Rationale for adding feature
8PO asked us to do this and it's annoying. The default is 12:40, but they seem to be using differing times.

## Suggested implementation(s) (if applicable)
- This would probably be as easy as adding an <input> to the template and then doing something with that input in the view.

### PR DESCRIPTION
**Pull request by nkenschaft**
closes #1097

### CODE REVIEW

**Review by Laur04 — CHANGES_REQUESTED** [review]
In addition to fixing the broken tests, please change the commit message, replacing "added" with "add". Commit messages in general, but definitely for Ion, should be in the format of an imperative. For example, "fixed issue X", not "fixing issue X", "fixed issue X" or "will fix issue X"

**Comment by nkenschaft** [comment]
> In addition to fixing the broken tests, please change the commit message, replacing "added" with "add". Commit messages in general, but definitely for Ion, should be in the format of an imperative. For example, "fixed issue X", not "fixing issue X", "fixed issue X" or "will fix issue X"

Done.

File: intranet/apps/eighth/views/admin/blocks.py [hunk 1]
```diff
@@ -35,11 +36,18 @@ def add_block_view(request):
...
     show_letters = None
+    signup_time = datetime.time(12, 40)
 
     if "date" in request.GET:
         date = request.GET.get("date")
     if "date" in request.POST:
         date = request.POST.get("date")
+
+    if "signup_time" in request.GET:
+        signup_time = request.GET.get("signup_time")
```
Comment (Laur04) [inline intranet/apps/eighth/views/admin/blocks.py:47]:
```suggestion
```
You don't need this, as `signup_time` only matters on the POST request.

File: intranet/apps/eighth/views/admin/blocks.py [hunk 2]
```diff
@@ -35,11 +36,18 @@ def add_block_view(request):
...
+    if "signup_time" in request.POST:
+        signup_time = request.POST.get("signup_time")
```
Comment (Laur04) [inline intranet/apps/eighth/views/admin/blocks.py:49]:
```suggestion
    signup_time = request.POST.get("signup_time") if "signup_time" in request.POST else datetime.time(12, 40)
```

File: intranet/apps/eighth/views/admin/blocks.py [hunk 3]
```diff
@@ -35,11 +36,18 @@ def add_block_view(request):
 
     date = None
     show_letters = None
+    signup_time = datetime.time(12, 40)
```
Comment (Laur04) [inline intranet/apps/eighth/views/admin/blocks.py:39]:
```suggestion
```

File: intranet/apps/eighth/views/admin/blocks.py [hunk 4]
```diff
@@ -57,8 +65,8 @@ def add_block_view(request):
                 if not ltr:
                     continue
                 if ltr not in current_letters:
-                    EighthBlock.objects.create(date=fmtdate, block_letter=ltr)
-                    messages.success(request, "Successfully added {} Block on {}".format(ltr, fmtdate))
+                    EighthBlock.objects.create(date=fmtdate, block_letter=ltr, signup_time=signup_time)
+                    messages.success(request, "Successfully added {} Block on {} at {}".format(ltr, fmtdate, signup_time))
```
Comment (Laur04) [inline intranet/apps/eighth/views/admin/blocks.py:69]:
```suggestion
                    messages.success(request, "Successfully added {} Block on {} with signups shown to students as closing at {}".format(ltr, fmtdate, signup_time))
```

File: intranet/templates/eighth/admin/add_block.html [hunk 5]
```diff
@@ -46,6 +46,12 @@
                 <input type="text" name="blocks" size="12" maxlength="10">
             </td>
         </tr>
+        <tr>
+            <th><label>Set time:</label></th>
```
Comment (Laur04) [inline intranet/templates/eighth/admin/add_block.html:50]:
```suggestion
            <th><label>Set signup locking time:</label></th>
```

File: intranet/templates/eighth/admin/add_block.html [hunk 6]
```diff
@@ -46,6 +46,12 @@
...
+            <td>
+                <input type="time" name="signup_time" size="12" maxlength="10">
```
Comment (Laur04) [inline intranet/templates/eighth/admin/add_block.html:52]:
```suggestion
                <input type="time" name="signup_time" size="12" maxlength="10" value="12:40">
```

File: intranet/templates/eighth/admin/add_block.html [hunk 7]
```diff
@@ -106,8 +112,12 @@
         }
     }
     console.debug(txt);
+
+    var time = $("input[name=signup_time][type=time]").val();
+    console.debug(time, "time");
```
Comment (Laur04) [inline intranet/templates/eighth/admin/add_block.html:117]:
```suggestion
```

**Review by Laur04 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**